py-polars/polars/dataframe/frame.py

```diff
@@ -3503,23 +3503,48 @@ def write_database(
         else:
             raise ValueError(f"engine {engine!r} is not supported")
 
+    @overload
+    def write_delta(
+        self,
+        target: str | Path | deltalake.DeltaTable,
+        *,
+        mode: Literal["merge"],
+        overwrite_schema: bool = ...,
+        storage_options: dict[str, str] | None = ...,
+        delta_merge_options: dict[str, Any],
+    ) -> deltalake.table.TableMerger:
+        ...
+
+    @overload
     def write_delta(
         self,
         target: str | Path | deltalake.DeltaTable,
         *,
-        mode: Literal["error", "append", "overwrite", "ignore"] = "error",
+        mode: Literal["error", "append", "overwrite", "ignore"] = ...,
```

Comment:
Could you move this overload definition above the other one? Not 100% sure if it's required by mypy but at least it's the convention in our code to have the overload definition for the default argument first.

Reply:
Sure! Mypy passed but it makes sense to keep the proper order